src/Build/Logging/FancyLogger/FancyLoggerBuffer.cs

```diff
@@ -0,0 +1,262 @@
+    /// </summary>
+    public class FancyLoggerBufferLine
+    {
+        private static int counter = 0;
+        public int Id;
+        public string Text;
+        public bool IsHidden;
+        public int IdentationLevel = -1;
+        public FancyLoggerBufferLine()
+        {
+            Id = counter++;
+            Text = String.Empty;
+        }
+        public FancyLoggerBufferLine(string text)
+        {
+            Id = counter++;
+            Text = text;
+        }
+        public FancyLoggerBufferLine(string text, int identationLevel) : this(text)
+        {
+            IdentationLevel = identationLevel;
+        }
+        public void Hide()
+        {
+            IsHidden = true;
+        }
+        public void Unhide()
+        {
+            IsHidden = false;
+        }
+        public int GetIndex()
+        {
+            return FancyLoggerBuffer.GetLineIndexById(Id);
+        }
+    }
+
+    /// <summary>
+    /// Buffer manager for FancyLogger
+    /// </summary>
+    internal static class FancyLoggerBuffer
+    {
+        // Status
+        public static bool AutoScrollEnabled { get; private set; }
+        public static bool IsTerminated { get; private set; }
+        public static int CurrentTopLineIndex { get; private set; }
+        public static int Height { get { return Console.WindowHeight; } }
+        // Lines to be presented by the buffer
+        private static List<FancyLoggerBufferLine> lines = new();
+        public static void Initialize()
+        {
+            // Setup event listeners
+            Task.Run(() =>
+            {
+                while (true)
+                {
+                    if (IsTerminated) return;
+                    switch (Console.ReadKey().Key)
+                    {
+                        case ConsoleKey.UpArrow:
+                            ScrollToLine(CurrentTopLineIndex - 1);
+                            break;
+                        case ConsoleKey.DownArrow:
+                            ScrollToLine(CurrentTopLineIndex + 1);
+                            break;
+                        case ConsoleKey.Home:
+                            ScrollToLine(0);
+                            break;
+                        case ConsoleKey.End:
+                            ScrollToEnd();
+                            break;
+                        case ConsoleKey.Spacebar:
+                        case ConsoleKey.Escape:
+                            ToggleAutoScroll();
+                            break;
+                    }
+                }
+            });
+            // Switch to alternate
+            Console.Write(ANSIBuilder.Buffer.UseMainBuffer());
+            Console.Write(ANSIBuilder.Buffer.UseAlternateBuffer());
+            // Settings
+            AutoScrollEnabled = true;
+            // Render contents
+            WriteTitleBar();
+            WriteFooter("This is an empty footer haha");
```

Comment:
```suggestion
            WriteFooter("This is an empty footer hoho");
```

Gotta get in the Christmas spirit!